src/cli/argument-parser.js

```diff
@@ -124,8 +139,10 @@ export default class CLIArgumentParser {
     _parseFilteringOptions () {
         this.opts.testGrep    = CLIArgumentParser._optionValueToRegExp('--test-grep', this.opts.testGrep);
         this.opts.fixtureGrep = CLIArgumentParser._optionValueToRegExp('--fixture-grep', this.opts.fixtureGrep);
+        this.opts.testMeta    = CLIArgumentParser._optionValueToJson('--test-meta', this.opts.testMeta);
+        this.opts.fixtureMeta = CLIArgumentParser._optionValueToJson('--fixture-meta', this.opts.fixtureMeta);
```

Comment:
Good catch, i'm not a windows guy so i really dind't knew that.

The problem with using a `key=value,[key2=value,..]` syntax is that all types would be string (ie, boolean and numbers will be transform to strings) and that will make ìsMatch`to fail.

Eg:
With a test:
```js
test
  .meta('key',  1)
  .meta('mobile',  true);
```
`key=1,mobile=true` will be transformed to `{ key: '1', mobile: 'true' }` and the `isMatch` check will fail as the test meta was defined as a number and boolean.

In addition, i think that it will be useful to allow users to match elements within a metadata array:
```js
test
  .meta('platforms', ['mobile', 'desktop']);
```

What about having the following syntax: `string=value:array=['string',true]:bool=true` (`[` and `]`are actual characters, the idea is to allow passing an array)

With the following parser:
```js
value
    .split(':')
    .reduce((map, rawPair) => {
      const [key, value] = rawPair.split('=');
      try {
        map[key] = JSON.parse(value.replace(/'/g, '"'));
      } catch(e) {
        map[key] = value;
      }
      return map;
    }, {});
```
And change the `isMatch` call to:

```js
_.isMatchWith(
    testMeta,
    parsedMeta,
    (meta, parsed) => {
      if (_.isArray(meta)) {
        return parsed.length === _.intersection(meta, parsed).length;
      }
    }
  );
```

This will allow to pass strings without the double quotes, fix the different type issues and also allow to match elements within a metadata array.

[Examples can be found here](https://gist.github.com/NickCis/5e1cec0e4ea9f8c5db15831ddaccda2c)

Although, i don't see as something intuitive to use the two dots character (`:`) as a separator, perhaps would it be better to use the ampersand (`&`)?. Moreover, this parser lacks of a proper validation, a naive validation could be done by checking if the parsed object isn't empty, but this won't catch errors of the `JSON.parse` (because this failed parsing could be a valid string).

What will be better to do?:
1. Keep the raw json
2. Implement the proposed solution: `key=value,[key2=value2,...]`
    1. Without type guessing (all arguments will be transformed to strings and in order to match, test's metadata should be defined as string)
    2. Respect test's metadata types: while doing the `isMatch` try to type cast the parsed argument to the metadata type (casting errors will be taken as a negative match)
    3. Use a naive type guessing: run `JSON.parse` if it works, leave the parsed value, if not, use the string
3. Implement the alternative solution (in order to allow array's element matching):
    1. With `:` separator
    2. with `&` separator

Reply:
@AndreyBelym @VasilyStrelyaev @miherlosev any thoughts about this?

Reply:
I think that type erasure in this case is not a problem, because all meta values must be strings, take a look at the [metadata docs](http://devexpress.github.io/testcafe/documentation/test-api/test-code-structure.html#specifying-testing-metadata). I think that TestCafe should allow to specify values that can be automatically coerced to the string type, but all other values must be rejected. Unfortunately, currently we don't have any validation for metadata values, but it's a topic for another improvement.

Talking about your second idea (allow to specify arrays as metadata values), I also think that it's better to keep metadata values to be only strings, but allow to use special glob characters ('*', '?', etc.) in metadata filter. But IMHO, it's also topic for another discussion, and I would like to see this improvement in a separate PR.